### PR TITLE
fixed/better parsing of connect cookie for socketId

### PR DESCRIPTION
**Pull request by kulicuu**
Something required a parameter change here.

### CODE REVIEW

**Comment by hulmgulm** [comment]
While trying to figuring out our problems with socketstream (https://groups.google.com/forum/#!topic/socketstream/lCZRU2whpQw) we came across this code too. Why do you expect the session cookie to be the first or third cookie? Why not look for the cookie name? Then it doesn't matter in which order the cookies are set. According to the error message at the end of this function, the cookie should always be connect.sid.

We were setting additional cookies a few days ago which led to strange problems like "undefinded" or "true" as session id. Or to constantly changing sessions ids.

**Comment by kulicuu** [comment]
Why do you expect the session cookie to be the first or third cookie? Why
not look for the cookie name?

Yes great question. I was actually just doing a version that looked for the
cookie name by for loop, but it wasn’t working for some reason, so wanted
to add this his is a quick patch. The reason it’s looking for the first or
third is that when I was last doing SocketStream development (two months
ago), on the first uncached load the cookie was different than on
subsequent runs. I’ll have the name-checking version working a bit later.
Sorry, I’m quickly getting back up to speed here, but just second day.
​

**Comment by kulicuu** [comment]
I need to figure out if something changed with the some of the engine dependencies or some intervening code.  (It will be slowly, slowly while I review two months of commits and changes.)

**Comment by kulicuu** [comment]
Okay it's name detection now.  Works for me on Firefox, Chrome and Chromium. 
Maybe it's the browsers were changing in the meantime, I have to go look at engineio client.  Anyways I think it's good now.

**Comment by hulmgulm** [comment]
Looks good with the exception of these "var c = " statements

**Comment by kulicuu** [comment]
fixed.

File: lib/websocket/transports/engineio/index.js [hunk 1]
```diff
@@ -9,13 +9,13 @@ var openSocketsById = {};
...
-    var raw= socket.request.headers.cookie.split(';');
-    if (raw.length === 1) {
-      socket.sessionId = (raw[0].split('='))[1];
-    }
-    else {
-      socket.sessionId = (raw[1].split('='))[1];
-    }
+    socket.request.headers.cookie.split(';').some(function(item) {
+        var item_di = item.split('=');
+        if (item_di[0].trim() == 'connect.sid') {
```
Comment (thepian) [inline lib/websocket/transports/engineio/index.js]: shouldn't this be using configuration ?

Comment (kulicuu) [inline lib/websocket/transports/engineio/index.js]: It's a bit different now that I've fixed #543 , in that the cookieParser secret is injected.  Are there more config variables that should be brought in here ?

**Pull request merged**